modular_lumos/code/modules/_lumos/lobby_cam.dm

```diff
@@ -0,0 +1,155 @@
+			P.client.screen -= thescreen
+		if(P.client.eye != src)
+			continue
+		P.client.perspective = MOB_PERSPECTIVE
+		P.client.eye = P.client.mob
+		P.see_in_dark = initial(P.see_in_dark)
+		P.see_invisible = initial(P.see_invisible)
+		P.sight = initial(P.sight)
+
+/obj/new_player_cam/New()
+	for(var/turf/open/floor/T in world)
+		if(!is_station_level(T.z))
+			continue
+		camturfs += T
+	thescreen = new()
+	thescreen.icon = 'icons/effects/ss13_dark_alpha6.dmi'
+	thescreen.icon_state = "0"
+	thescreen.screen_loc = "SOUTH,WEST to NORTH,EAST"
+	thescreen.mouse_opacity = 0
+	thescreen.layer = 20
+	thescreen.plane = 100
+	thescreen.color = "black"
+	spawn(0)
+		while(1)
+			if(SSticker.current_state > GAME_STATE_PREGAME||!camturfs.len)
+				unlock_eyes_from_cam()
+				loc = null
+				sleep(5)
+				continue
+			if(SSticker.current_state < GAME_STATE_PREGAME)
+				unlock_eyes_from_cam()
+				sleep(5)
+				continue
+			var/list/destinations = list()
+			var/turf/start = pick(camturfs)
+			while(get_dist(start,previousstart) < 40)
+				start = pick(camturfs)
+			for(var/Dir in GLOB.alldirs)
+				var/spacecrossed = 0
+				var/turf/current = start
+				for(var/i=tiles_moved_per_shot,i>0,i--)
+					var/turf/thestep = get_step(current,Dir)
+					if(thestep)
+						destinations["[Dir]"] = get_dist(start,thestep)
+						current = thestep
+						if(istype(thestep,/turf/open/space) && !thestep.contents.len)
+							spacecrossed++
+							if(spacecrossed >= reset_shot_if_no_floors_in_this_range)
+								break
+			var/finaldir
+			if(destinations.len)
+				var/farthest = 0
+				var/list/maxdirs = list()
+				for(var/Dir in destinations)
+					var/thedir = text2num(Dir)
+					var/thedist = destinations[Dir]
+					if(!finaldir||!farthest)
+						farthest = thedist
+						finaldir = thedir
+					if(thedist > farthest)
+						farthest = thedist
+						maxdirs = list()
+					if(thedist >= farthest)
+						maxdirs += thedir
+				if(maxdirs.len)
+					finaldir = pick(maxdirs)
+			else
+				finaldir = pick(GLOB.alldirs)
+			loc = start
+			lock_eyes_to_cam()
+			var/alphasync = 1
+			spawn(0)
+				for(var/i=0,i<=6,i++)
+					if(!alphasync)
+						break
+					thescreen.icon_state = "[i]"
+					sleep(1)
+				thescreen.icon_state = "6"
+			sleep(scroll_speed)
+			for(var/i=tiles_moved_per_shot,i>0,i--)
+				if(SSticker.current_state >= GAME_STATE_PLAYING)
+					qdel(src)
+					return
+				var/turf/newturf = get_step(loc,finaldir)
+				if(newturf)
+					loc = newturf
+					for(var/mob/dead/new_player/P in world) // Parallax Update - Killing Torcher
+						if (!P || !P.client || P.client.eye != src || !P.hud_used || P.client.byond_version < 511)
+							continue
+						P.hud_used.update_parallax()
+
+					sleep(scroll_speed)
+				else
+					sleep(scroll_speed)
+					break
+			alphasync = 0
+			for(var/i=6,i>0,i--)
+				thescreen.icon_state = "[i]"
+				sleep(1)
+			thescreen.icon_state = "0"
+			sleep(1)
+
+/proc/message_falaskian(message)
```

Comment:
huh

Reply:
a bone of the port. I can remove it lmao